### PR TITLE
Updated-SharePoint Framework development with SharePoint Server 2019 …

### PR DESCRIPTION
**Pull request by nimishasatapathy**
…& SharePoint Server SE

Category
[X] Content fix

Related issues
No related issues

What's in this Pull Request?
The main purpose of this PR is to address the difference in SPFx dev between for SharePoint online and for on-premises.

### CODE REVIEW

File: docs/spfx/set-up-your-development-environment.md [hunk 1]
```diff
@@ -188,4 +188,4 @@ If your development environment is behind a corporate proxy, you need to configu
 
 - [SharePoint Framework development tools and libraries compatibility](compatibility.md)
 - [SharePoint Framework development with SharePoint Server 2016 Feature Pack 2](sharepoint-2016-support.md)
-- [SharePoint Framework development with SharePoint Server 2019](sharepoint-2019-support.md)
+- [SharePoint Framework development with SharePoint Server 2019](sharepoint-2019-and-subscription-edition-support.md)
```
Comment (andrewconnell) [inline docs/spfx/set-up-your-development-environment.md:191]: Shouldn't the title of this page change if it's not just about SP2019?

File: docs/spfx/sharepoint-2019-and-subscription-edition-support.md [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+> [!IMPORTANT]
+> SharePoint Server Subscription Edition (SE) has all the same dependencies and requirements for the SharePoint Framework as SharePoint Server 2019.
+
+
+## Which version of the SharePoint Framework to use
+
+Because SharePoint Online and SharePoint Server 2019 have different release cycles for new capabilities, they also have different capabilities when it comes to the SharePoint Framework. SharePoint Online always uses the latest version of the SharePoint Framework, but SharePoint Server 2019 only supports the version that matches the server-side dependencies of the deployed packages.
+
+> [!IMPORTANT]
+>SharePoint Server 2019 supports SharePoint Framework client-side web parts hosted on classic or modern SharePoint pages built by using the SharePoint Framework v1.4.1.
```
Comment (andrewconnell) [inline docs/spfx/sharepoint-2019-and-subscription-edition-support.md:21]: Missing leading space after the `>`

File: docs/spfx/sharepoint-2019-and-subscription-edition-support.md [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+
+The [Team-based development on the SharePoint Framework](team-based-development-on-sharepoint-framework.md) guidance document includes different options for development environment setup including when you might need to support multiple SharePoint Framework versions.
+
+### Node.js, gulp-cli, and Yeoman versions
+
+The dependencies for SPFx v1.4.1 frameworks, tools, and the associated versions don't match the same dependency matrix for the latest versions of SharePoint Framework. In these cases, you may need to install specific versions of the tools. Because SPFx v1.4.1 is supported on Node.js v6 and Node.js v8 by default, you need to ensure you have a supported version of Node.js installed as well as an older version of Gulp CLI and Yeoman.
+
+> [!NOTE]
+   >- The Gulp team introduced a separate package, [gulp-cli](https://www.npmjs.com/package/gulp-cli), that should be installed globally. It can be used by projects that use either Gulp v3 & Gulp v4.
```
Comment (andrewconnell) [inline docs/spfx/sharepoint-2019-and-subscription-edition-support.md:40]: Incorrect markdown formatting... all `>` should be left aligned. For reference, [see this example](https://github.com/SharePoint/sp-dev-docs/blob/main/docs/business-apps/power-automate/sharepoint-connector-actions-triggers.md?plain=1#L183-L188).

File: docs/spfx/sharepoint-2019-and-subscription-edition-support.md [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+
+      “dependencies”:{ 
+         “graceful-fs”: { 
+                “version”: “4.2.2” 
+          } 
+       } 
+    }
+```   
+
+2. Install global dependencies.<br>
```
Comment (andrewconnell) [inline docs/spfx/sharepoint-2019-and-subscription-edition-support.md:64]: Incorrect use of markdown numbered bullets. These should always be `1.` and let the rendering engine render the numbers for you. [See this for reference](https://github.com/SharePoint/sp-dev-docs/blob/main/docs/apis/migration-api-overview.md?plain=1#L574-L584)

File: docs/spfx/sharepoint-2019-and-subscription-edition-support.md [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+      “dependencies”:{ 
+         “graceful-fs”: { 
+                “version”: “4.2.2” 
+          } 
+       } 
+    }
+```   
+
+2. Install global dependencies.<br> 
+   `npm install gulp-cli@2.3.0 --global`<br>
```
Comment (andrewconnell) [inline docs/spfx/sharepoint-2019-and-subscription-edition-support.md:65]: Two issues with this one:

1. Don't use HTML tags (`<br>` in this case) in markdown files. This is unnecessary.
1. These are code listings that should be fenced code blocks, not individual lines of code.

File: docs/spfx/sharepoint-2019-and-subscription-edition-support.md [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+
+To create SharePoint Foundation Subscription Settings Service, use the following PowerShell command:
+
+```powershell
+$sa = New-SPSubscriptionSettingsServiceApplication -ApplicationPool $applicationPoolName -Name $serviceApplicationName -DatabaseName $dataBaseName 
+
+New-SPSubscriptionSettingsServiceApplicationProxy -ServiceApplication $sa
+```
+
+### Prepare .sppkg package
```
Comment (andrewconnell) [inline docs/spfx/sharepoint-2019-and-subscription-edition-support.md:97]: Same issues as mentioned above (these should be fenced code blocks, not inline code, & shouldn't use HTML tags)

File: docs/spfx/sharepoint-2019-and-subscription-edition-support.md [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+1. Package the solution.<br>
+   `gulp package-solution --ship`
+
+   Verify SharePoint Framework web part on local SharePoint workbench. 
+
+### Create and configure App Catalog site
+
+To create and configure App Catalog site, follow these steps:
+
+1. From the **Central Administration** site, go to **Apps** and then select **Manage App Catalog**.
```
Comment (andrewconnell) [inline docs/spfx/sharepoint-2019-and-subscription-edition-support.md:111]: Same as previous issue with numbered bullet points in markdown

File: docs/spfx/compatibility.md [hunk 8]
```diff
@@ -24,7 +24,7 @@ SharePoint Online always uses the latest version of the SharePoint Framework, bu
 For more information about SharePoint Framework development with SharePoint 2016 Feature Pack 2 and SharePoint 2019, see:
 
 - [SharePoint Framework development with SharePoint 2016 Feature Pack 2](sharepoint-2016-support.md)
-- [SharePoint Framework development with SharePoint Server 2019](sharepoint-2019-support.md) - this guidance applies also for SharePoint Server Subscription Edition
+- [SharePoint Framework development with SharePoint Server 2019](sharepoint-2019-and-subscription-edition-support.md) - this guidance applies also for SharePoint Server Subscription Edition
```
Comment (andrewconnell) [inline docs/spfx/compatibility.md:27]: Shouldn't the title of this page change if it's not just about SP2019?

File: docs/spfx/supported-extensibility-platforms-overview.md [hunk 9]
```diff
@@ -24,7 +24,7 @@ Learn more about using SPFx on SharePoint Online: [SPFx support for SharePoint O
 
 SharePoint Server 2019 includes support for the SPFx up to version v1.4.1.
 
-Learn more about using SPFx on SharePoint Server 2019: [SPFx support for SharePoint Server 2019](sharepoint-2019-support.md).
+Learn more about using SPFx on SharePoint Server 2019: [SPFx support for SharePoint Server 2019](sharepoint-2019-and-subscription-edition-support.md).
```
Comment (andrewconnell) [inline docs/spfx/supported-extensibility-platforms-overview.md:27]: Shouldn't the title of this page change if it's not just about SP2019?

File: docs/spfx/sharepoint-2019-and-subscription-edition-support.md
Comment (andrewconnell) [inline docs/spfx/sharepoint-2019-and-subscription-edition-support.md:1]: Mentioned this in previous PR's on this topic, but is there a reason why we need to duplicate the steps for creating an app catalog when that's already covered in docs by other teams, such as:

- https://learn.microsoft.com/en-us/sharepoint/administration/configure-an-environment-for-apps-for-sharepoint
- https://learn.microsoft.com/en-us/sharepoint/administration/manage-the-app-catalog

If there's a reason why this needs to be in this doc, then that's fine, but keep in mind the target audience for these docs are *developers*, not *administrators* who will manage the SharePoint deployments & installations.

**Review by andrewconnell — CHANGES_REQUESTED** [review]

**Comment by nimishasatapathy** [comment]
@andrewconnell , i have made the changes suggested by you. Please review and let me know if i have to make any more further changes.

Thanks!

File: docs/spfx/set-up-your-development-environment.md (hunk 1 above)
Comment (nimishasatapathy) [reply docs/spfx/set-up-your-development-environment.md]: Resolved

File: docs/spfx/sharepoint-2019-and-subscription-edition-support.md [hunk 10]
```diff
@@ -0,0 +1,111 @@
...
+> - Learn more about the gulp-cli here: [gulpjs/gulp/#2324](https://github.com/gulpjs/gulp/issues/2324).
+
+Microsoft recommends using the most recent version of the Yeoman generator for the SharePoint Framework [@microsoft/generator-sharepoint](https://www.npmjs.com/package/@microsoft/generator-sharepoint) that supports creating on-premises projects: SPFx v1.10.0.  
+
+> [!IMPORTANT]
+> The Yeoman generator for the SharePoint Framework, starting with v1.13.0, only supports projects for SharePoint Online. Learn more about this change in the [SharePoint Framework v1.13 release notes](release-1.13.md). However, SPFx 1.4.1 is  supported on Node.js v6 and Node.js v8 by default. Therefore, you need to get the latest version of the Yeoman generator for the SharePoint Framework (v1.10.0) that works on the same version of Node.js (v6 or v8) that SPFx v1.4.1 is supported on. Solution structure is created then with the v1.4.1 version packages as long as you select the environmen target properly in the Yeoman generator flow.
+
+### Prepare the environment for SharePoint Framework development
+
+1. Install [Node.js v8.17.0](https://nodejs.org/download/release/v8.17.0/).<br>
```
Comment (andrewconnell) [inline docs/spfx/sharepoint-2019-and-subscription-edition-support.md:51]: HTML tags don't belong in markdown files. There's a way to enforce a line break in a bullet list.

**Pull request merged**